### PR TITLE
Optimizations for Solver.Solve

### PR DESCRIPTION
**Pull request by weihesdlegend**
## Description
currently to solve each request, the solver queries multiple times of Redis sequentially. This is not very efficient considering the distributed system environment in production,

## Solution description
Use go routines to concurrently obtain `slotSolution` results

## Covered E2E tests
Added new unit tests and manually tested

## Final Checks
- [x] Have you removed commented code ?
- [x] Have you used gofmt to format your code ?
- [x] You are using approved terminology
- [x] You have added unit tests

### CODE REVIEW

File: test/redis_client_mocks/get_multi_slot_solutions_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		Country: "USA",
+	}
+	cacheResponse2 := iowrappers.SlotSolutionCacheResponse{}
+	cacheResponse2.SlotSolutionCandidate = make([]iowrappers.SlotSolutionCandidateCache, 1)
+	cacheResponse2.SlotSolutionCandidate[0].PlaceIds = []string{"11","22","33"}
+
+	RedisClient.CacheSlotSolution(cacheRequest2, cacheResponse2)
+
+	cacheResponses := []iowrappers.SlotSolutionCacheResponse{cacheResponse1, cacheResponse2}
+	multiSlotSolutions := RedisClient.GetMultiSlotSolutions([]iowrappers.SlotSolutionCacheRequest{cacheRequest1, cacheRequest2})
```
Comment (sananand007) [inline test/redis_client_mocks/get_multi_slot_solutions_test.go:31]: @weihesdlegend is this function Predefined or you defined is somewhere ?

Comment (weihesdlegend) [reply test/redis_client_mocks/get_multi_slot_solutions_test.go:31]: `GetMultiSlotSolutions` is defined in the `iowrappers` package

Comment (sananand007) [reply test/redis_client_mocks/get_multi_slot_solutions_test.go:31]: Cool I just wanted to understand, this test looks like a legacy test. Is it a new addition for this func ?

**Comment by weihesdlegend** [comment]
@sananand007 nope, it's a new test for the new function `GetMultiSlotSolutions`

**Pull request merged**